### ISSUE TITLE
Can not catch ocelot exception

### ISSUE THREAD

**Issue by Tingtzong** [ThreeMammals/Ocelot#1210]
## Expected Behavior / New Feature
when an exception (404 NotFound exception) occur in ocelot. when we  use try-catch try to catch this exception, we should be able catch this exception.

## Actual Behavior / Motivation for New Feature
 ocelot not throw this exception, swallowed, so the try-catch did not work. 

## Steps to Reproduce the Problem

  1. add a handler to ocelot.  add try-catch to this handler
  2. when an exception happen, try-catch code will be ignore, and can not get any exception.

## Specifications

  - Version: lastest
  - Platform: .net core
  - Subsystem:

**Comment by neetra** [comment]
@Tingtzong 
 Could you please share the sample code?

**Comment by raman-m** [comment]
@Tingtzong 
> ## Actual Behavior / Motivation for New Feature
> ocelot not throw this exception, swallowed, so the try-catch did not work. 

Exactly! Have you ever heard or read about [Error Status Codes](https://ocelot.readthedocs.io/en/latest/features/errorcodes.html) feature?
We have a global Exception handler which prevents Ocelot instance failing/hanging because of 5xx errors in downstream services.
But we don't consider this design good because we override status and some headers. See notes about [Design](https://ocelot.readthedocs.io/en/latest/features/errorcodes.html#design)

Please develop [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html) to catch exceptions in and read original status of downstream service response.
A delegating handler is exactly right place to catch exception, but we recommend to re-throw it!

Also, `PreErrorResponderMiddleware` injection should help you too. See [Middleware Injection](https://ocelot.readthedocs.io/en/latest/features/middlewareinjection.html) plz!
